broker.go

```diff
@@ -36,14 +38,26 @@ func (br *defaultBroker) Publish(pub Sender, msg *Publish) {
 		ArgumentsKw: msg.ArgumentsKw,
 		Details:     make(map[string]interface{}),
 	}
+subscriber:
 	for id, sub := range br.routes[msg.Topic] {
+		// don't send event to publisher
+		if sub == pub {
+			continue
+		}
+
+		subOptions := br.options[msg.Topic][id]
+		for option, pubValue := range msg.Options {
+			if subValue, ok := subOptions[option]; ok {
+				if subValue != pubValue {
```

Comment:
Hmm, yeah after my last set of changes I completely understand what this entails.  Equality is fine.  ;)